test/components/NavBar/NavBarSkipLink.test.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { BrowserRouter } from 'react-router-dom';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { expect, describe, it } from 'vitest';
+import { NavBarSkipLink } from '@components/NavBar';
+
+describe('NavBarSkipLink', () => {
+  it('renders', () => {
+    render(
+      <BrowserRouter>
+        <NavBarSkipLink />
+      </BrowserRouter>
+    );
+    const skipLink = screen.getByText('Skip to main content');
+    expect(skipLink).not.toBeNull();
+  });
+
+  it('moves the button in and out of view', async () => {
+    render(
+      <BrowserRouter>
+        <NavBarSkipLink />
+      </BrowserRouter>
+    );
+    const skipLink = screen.getByText('Skip to main content');
+
+    const initialOpacity = window.getComputedStyle(skipLink).opacity;
+    const initialTransform = window.getComputedStyle(skipLink).transform;
+
+    expect(initialOpacity).toBe('0');
+    expect(initialTransform).toBe('translateY(-100%)');
+
+    await userEvent.tab();
+
+    await waitFor(() => {
+      const afterTabKeyPressOpacity = window.getComputedStyle(skipLink).opacity;
+      const afterTabKeyPressTransform = window.getComputedStyle(skipLink).transform;
+
+      expect(afterTabKeyPressOpacity).toBe('1');
+      expect(afterTabKeyPressTransform).toBe('translateY(0%)');
+    });
+
+    await userEvent.click(skipLink);
+
+    await waitFor(() => {
+      const afterTabKeyPressOpacity = window.getComputedStyle(skipLink).opacity;
+      const skipLinkAfterClick = window.getComputedStyle(skipLink).transform;
+      // wait for animation to finish
+      setTimeout(() => {
```

Comment:
It think we could just check to see if the setTimeout which hosts the animation gets triggered? I'm not too sure with regards to animation testing.

Reply:
Or maybe if any of our testing libraries have an ability to check if an element is in viewport, instead of just checking if element exists on DOM?   setup so element is not in viewport, then after click check that it is...  not sure.

Reply:
Might be able to use `fireEvent.animationEnd` https://klaviyo.tech/hitting-a-moving-target-testing-javascript-animations-in-react-with-jest-8284a530a35a#4a82

```tsx
await fireEvent.animationEnd(skipLink);

// do tests here
```